### PR TITLE
Formik

### PR DESCRIPTION
**Pull request by Racenna**

### CODE REVIEW

File: src/components/Profile/MyPosts/MyPostForm/MyPostsForm.jsx [hunk 1]
```diff
@@ -1,32 +1,41 @@
...
-        <img src={send} alt='send svg' />
-      </button>
-    </form>
+    <Formik
+      initialValues={{
+        post: '',
+      }}
+      validate={postFormValidate}
+      onSubmit={(values, { resetForm }) => {
+        if (values.post.length === 0 || values.post.trim() === '') return;
```
Comment (worksnk) [inline src/components/Profile/MyPosts/MyPostForm/MyPostsForm.jsx:15]: this should be part of your validation

File: src/components/Profile/MyPosts/MyPostForm/MyPostsForm.jsx [hunk 2]
```diff
@@ -1,32 +1,41 @@
...
-        name='post'
-        placeholder='Type text'
-        validate={[maxLength100]}
-      />
-
-      <button>
-        <img src={send} alt='send svg' />
-      </button>
-    </form>
+    <Formik
```
Comment (worksnk) [inline src/components/Profile/MyPosts/MyPostForm/MyPostsForm.jsx:9]: use hook API

File: src/components/Profile/ProfileInfo/ProfileInfo.jsx [hunk 3]
```diff
@@ -54,8 +54,8 @@ const ProfileInfo = ({ isOwner, profile, status }) => {
         </div>
         {editMode ? (
           <ProfileDescriptionReduxForm
-            initialValues={profile}
-            onSubmit={onSubmit}
+            // initialValues={profile}
```
Comment (worksnk) [inline src/components/Profile/ProfileInfo/ProfileInfo.jsx:57]: always remove commented code

File: src/redux/profileReducer.js [hunk 4]
```diff
@@ -132,11 +132,12 @@ export const saveProfile = (profile) => async (dispatch, getState) => {
   if (response.resultCode === 0) {
     dispatch(getProfile(userId));
   } else {
-    const message =
-      response.messages.length > 0 ? response.messages[0] : 'Some error';
-    dispatch(stopSubmit('editProfile', { _error: message }));
-    return Promise.reject(message);
-    // dispatch(stopSubmit("editProfile", { contacts: { facebook: message } })); //TODO: parse string
+    // const message =
```
Comment (worksnk) [inline src/redux/profileReducer.js:135]: remove commented code

File: src/util/validator/validators.js [hunk 5]
```diff
@@ -8,3 +8,37 @@ export const maxLengthCreator = (maxLength) => (value) => {
...
+
+export const postFormValidate = (values) => {
+  const errors = {};
+  if (values.post.length > 100) {
+    errors.post = 'Must be 100 character or less';
+  }
+  return errors;
+};
+
+export const profileDescriptionValidate = (values) => {
```
Comment (worksnk) [inline src/util/validator/validators.js:20]: Use field level validation

Comment (worksnk) [reply src/util/validator/validators.js:20]: And read about validationSchema in formik documentation

File: src/components/Profile/ProfileInfo/ProfileDescription/ProfileDescriptionForm.jsx [hunk 6]
```diff
@@ -1,64 +1,85 @@
...
+                  type='text'
+                />
+                <ErrorMessage
+                  name='lookingForAJobDescription'
+                  component='div'
+                />
+              </div>
+              <div className={styles.edit_contacts}>
+                <span>Contacts:</span>
+                {Object.keys(values.contacts).map((key) => {
```
Comment (worksnk) [inline src/components/Profile/ProfileInfo/ProfileDescription/ProfileDescriptionForm.jsx:64]: read documentation for formik https://formik.org/docs/guides/arrays

**Review by worksnk — CHANGES_REQUESTED** [review]

File: src/components/Profile/MyPosts/MyPostForm/MyPostsForm.jsx (hunk 1 above)
Comment (Racenna) [reply src/components/Profile/MyPosts/MyPostForm/MyPostsForm.jsx:15]: forgot to remove

File: src/components/Profile/MyPosts/MyPostForm/MyPostsForm.jsx (hunk 2 above)
Comment (Racenna) [reply src/components/Profile/MyPosts/MyPostForm/MyPostsForm.jsx]: Replaced Formik with useFormik

File: src/components/Profile/ProfileInfo/ProfileInfo.jsx (hunk 3 above)
Comment (Racenna) [reply src/components/Profile/ProfileInfo/ProfileInfo.jsx:57]: Removed comments in code

File: src/redux/profileReducer.js (hunk 4 above)
Comment (Racenna) [reply src/redux/profileReducer.js:135]: Removed comments in code

File: src/components/Profile/ProfileInfo/ProfileInfo.jsx (hunk 3 above)
Comment (worksnk) [reply src/components/Profile/ProfileInfo/ProfileInfo.jsx:57]: But it's right here. It wasn't removed

File: src/redux/profileReducer.js (hunk 4 above)
Comment (worksnk) [reply src/redux/profileReducer.js:135]: same as before

File: src/components/Profile/ProfileInfo/ProfileDescription/ProfileDescriptionForm.jsx (hunk 6 above)
Comment (worksnk) [reply src/components/Profile/ProfileInfo/ProfileDescription/ProfileDescriptionForm.jsx:64]: read the doc and check if you can improve it

Comment (worksnk) [reply src/components/Profile/ProfileInfo/ProfileDescription/ProfileDescriptionForm.jsx:64]: Also you didn't change to formik hooks here. When I leave a comment I'm not going to comment each single time it happens. I expect you finding places like this and fixing them

Comment (Racenna) [reply src/components/Profile/ProfileInfo/ProfileDescription/ProfileDescriptionForm.jsx]: I must use useField here?

Comment (worksnk) [reply src/components/Profile/ProfileInfo/ProfileDescription/ProfileDescriptionForm.jsx]: Use yup. Read about how to work with arrays and check see how you can change your component

**Comment by worksnk** [comment]
@Racenna Don't forget to remove redux-form from the project. Remove it from dependecies

File: package.json [hunk 7]
```diff
@@ -22,7 +23,8 @@
     "redux-form": "^8.3.6",
```
Comment (worksnk) [inline package.json:23]: remove it

Comment (Racenna) [reply package.json]: removed

**Review by worksnk — APPROVED** [review]

**Pull request merged**